### PR TITLE
FIX: Minor fixes to plotting

### PR DESCRIPTION
**Pull request by Eric89GXL**
This allows both `colormap` and `clim` to be `'auto'`. This way I get a plot using `mne_analyze` colormap with this:
```
stc.plot(lims=dict(pos_lims=(10, 50, 90)))
```
and a plot with `hot` colormap I get this:
```
stc.plot(lims=dict(lims=(10, 50, 90)))
```

@wronk please have a look and see if you're happy.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
I also dropped the language about PySurfer 0.4 regarding views because it has been out for almost two years now.

**Comment by Eric89GXL** [comment]
cc @rkmaddox if you have time you could pull and test and see if it works for your code the way you expect

**Comment by Eric89GXL** [comment]
Open question: should the default for `stc.plot()` going forward be to use `mne_analyze` colormap with positive and negative limits, or `hot` colormap with only positive? I have more use cases and I think it's probably more common for `mne_analyze` to be the default, so I'd like to have it be that. This is a good opportunity to change it if we want to...

**Comment by wronk** [comment]
I think it comes down to what inverse method is used most often of MNE, dSPM, and sLORETA. Personally, `mne_analyze` would make more sense for me too.

File: mne/viz/_3d.py [hunk 1]
```diff
@@ -447,17 +456,19 @@ def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
...
+        0 and 255. If 'auto', either 'hot' or 'mne_analyze' will be chosen
+        based on whether 'lims' or 'pos_lims' are specified in `clim`.
     time_label : str
         How to print info about the time instant visualized.
     smoothing_steps : int
         The amount of smoothing
-    transparent : bool
+    transparent : bool | None
         If True, use a linear transparency between fmin and fmid.
+        None will choose automatically based on colormap type.
```
Comment (wronk) [inline mne/viz/_3d.py]: Can you copy/paste this new documentation and API into /mne/source_estimate.py so they're aligned?

**Comment by agramfort** [comment]
does it affect any of our examples?

File: mne/viz/_3d.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/viz/_3d.py]: I already did, I think, can you double-check?

**Comment by rkmaddox** [comment]
Yep @Eric89GXL, looks to do what it should do with properly interacting ``clim`` and ``colormap`` parameters. Thanks.

**Comment by Eric89GXL** [comment]
@agramfort the changes are designed to be backward-compatible, but the changes do allow us to simplify our examples. They have now been updated.

File: mne/viz/_3d.py (hunk 1 above)
Comment (wronk) [inline mne/viz/_3d.py]: Yep, looks fine -- temporary blindness I guess.

**Comment by agramfort** [comment]
thanks @Eric89GXL !

**Pull request merged**